discord/ui/select.py

```diff
@@ -251,77 +383,415 @@ def append_option(self, option: SelectOption) -> None:
+
+    Please note that if you use this in a private message with a user, no channels will be displayed to the user.
+
+    .. versionadded:: 2.1
+
+    Parameters
+    ------------
+    custom_id: :class:`str`
+        The ID of the select menu that gets received during an interaction.
+        If not given then one is generated for you.
+    channel_types: List[:class:`~discord.ChannelType`]
+        The types of channels to show in the select menu. Defaults to all channels.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 0 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not.
+    row: Optional[:class:`int`]
+        The relative row this select menu belongs to. A Discord component can only have 5
+        rows. By default, items are arranged automatically into those 5 rows. If you'd
+        like to control the relative positioning of the row then passing an index is advised.
+        For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    """
 
+    __item_repr_attributes__ = BaseSelect.__item_repr_attributes__ + ('channel_types',)
 
+    def __init__(
+        self,
+        *,
+        custom_id: str = MISSING,
+        channel_types: List[ChannelType] = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        disabled: bool = False,
+        row: Optional[int] = None,
+    ) -> None:
+        super().__init__(
+            self.type,
+            custom_id=custom_id,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            disabled=disabled,
+            row=row,
+            channel_types=[] if channel_types is MISSING else channel_types,
+        )
+
+    @property
+    def type(self) -> Literal[ComponentType.channel_select]:
+        return ComponentType.channel_select
+
+    @property
+    def channel_types(self) -> List[ChannelType]:
+        """List[:class:`~discord.ChannelType`]: A list of channel types that can be selected."""
+        return self._underlying.channel_types
+
+    @property
+    def values(self) -> List[Union[AppCommandChannel, AppCommandThread]]:
+        """List[Union[:class:`~discord.app_commands.AppCommandChannel`, :class:`~discord.app_commands.AppCommandThread`]]: A list of channels selected by the user."""
+        return super().values  # type: ignore
+
+
+@overload
 def select(
     *,
+    cls: Type[SelectT],
```

Comment:
```suggestion
    cls: Type[SelectT] = ...,
```

Reply:
This caused issues actually: 
```python
@overload
def select(
    *,
    cls: Type[SelectT] = ...,
    ...
)
```
```python
Type variable "SelectT" may go unsolved if caller supplies no argument for parameter "cls"
  Provide an overload that specifies the return type when the argument is not supplied
```

The appropriate solution I discovered was to set the default in this overload to `Select`. In my testing this caused no additional issues and solved the issue 👍
```python
@overload
def select(
    *,
    cls: Type[SelectT] = Select,
    ...
)